### ISSUE TITLE
The standalone pulsar runs sql-worker error with macOS.

### ISSUE THREAD

**Issue by warriorsfly** [apache/pulsar#15830]
I'm following the [guide](https://pulsar.apache.org/zh-CN/docs/next/sql-getting-started)，step 2, I mean the `./bin/pulsar sql-worker run`,then get errors below.
```zsh
openjdk 17.0.3 2022-04-19
OpenJDK Runtime Environment Homebrew (build 17.0.3+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.3+0, mixed mode, sharing)
```

2022-05-30T11:01:01.951+0800	INFO	main	com.google.inject.Guice	An exception was caught and reported. Message: java.lang.NoClassDefFoundError: Could not initialize class com.google.inject.internal.cglib.core.$MethodWrapper

java.lang.IllegalStateException: Unable to load cache item

​	at com.google.inject.internal.cglib.core.internal.$LoadingCache.createEntry(LoadingCache.java:79)

​	at com.google.inject.internal.cglib.core.internal.$LoadingCache.get(LoadingCache.java:34)

​	at com.google.inject.internal.cglib.core.$AbstractClassGenerator$ClassLoaderData.get(AbstractClassGenerator.java:119)

​	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.create(AbstractClassGenerator.java:294)

​	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.create(FastClass.java:65)

​	at com.google.inject.internal.BytecodeGen.newFastClassForMember(BytecodeGen.java:258)

​	at com.google.inject.internal.BytecodeGen.newFastClassForMember(BytecodeGen.java:207)

​	at com.google.inject.internal.ProviderMethod.create(ProviderMethod.java:69)

​	at com.google.inject.internal.ProviderMethodsModule.createProviderMethod(ProviderMethodsModule.java:327)

​	at com.google.inject.internal.ProviderMethodsModule.getProviderMethods(ProviderMethodsModule.java:135)

​	at com.google.inject.internal.ProviderMethodsModule.configure(ProviderMethodsModule.java:105)

​	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:347)

​	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:356)

​	at com.google.inject.spi.Elements.getElements(Elements.java:104)

​	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:137)

​	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:105)

​	at com.google.inject.Guice.createInjector(Guice.java:87)

​	at io.airlift.bootstrap.Bootstrap.initialize(Bootstrap.java:262)

​	at io.prestosql.server.PrestoServer.run(PrestoServer.java:111)

​	at io.prestosql.$gen.Presto_332____20220530_030059_1.run(Unknown Source)

​	at io.prestosql.server.PrestoServer.main(PrestoServer.java:72)

Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.google.inject.internal.cglib.core.$MethodWrapper

​	at com.google.inject.internal.cglib.core.$DuplicatesPredicate.evaluate(DuplicatesPredicate.java:104)

​	at com.google.inject.internal.cglib.core.$CollectionUtils.filter(CollectionUtils.java:52)

​	at com.google.inject.internal.cglib.reflect.$FastClassEmitter.<init>(FastClassEmitter.java:69)

​	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.generateClass(FastClass.java:77)

​	at com.google.inject.internal.cglib.core.$DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)

​	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.generate(AbstractClassGenerator.java:332)

​	at com.google.inject.internal.cglib.core.$AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:96)

​	at com.google.inject.internal.cglib.core.$AbstractClassGenerator$ClassLoaderData$3.apply(AbstractClassGenerator.java:94)

​	at com.google.inject.internal.cglib.core.internal.$LoadingCache$2.call(LoadingCache.java:54)

​	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)

​	at com.google.inject.internal.cglib.core.internal.$LoadingCache.createEntry(LoadingCache.java:61)

​	... 20 more

**Comment by boatrainlsz** [comment]
@warriorsfly maybe you should try a jdk that lower than 17, see [this](https://github.com/nroduit/Weasis/issues/201), after changing to jdk11, it works fine:
![image](https://user-images.githubusercontent.com/18243785/171874843-80a358dd-0ba7-4a96-8ef6-bbefd7d7142c.png)

**Comment by warriorsfly** [comment]
> @warriorsfly maybe you should try a jdk that lower than 17, see [this](https://github.com/nroduit/Weasis/issues/201), after changing to jdk11, it works fine:
> ![image](https://user-images.githubusercontent.com/18243785/171874843-80a358dd-0ba7-4a96-8ef6-bbefd7d7142c.png)

Hi,Dragon Boat Festival passed now. I tested, you are right, java11 works fine.
Em,at first my java version is 18(home-brew  upgraded),with the error info ，after reading the documents(see [this](https://pulsar.apache.org/docs/getting-started-standalone) in the get started,reinstall java at version 17, still error, so at that time, I think I don't need to test java11, because if it works, means the document need to fix this problem.

**Comment by lgxbslgx** [comment]
Such similar issues began at OpenJDK16 because of [JEP-396](https://openjdk.java.net/jeps/396). So you need to add something like `--add-opens java.base/java.lang=ALL-UNNAMED` to the options when using `java` command after JDK 16 (included).

Back to `sql-worker`, in order to solve @warriorsfly 's error of JDK17, you need to add the following two lines to the file `conf/presto/jvm.config`. Please note that you can't put these two lines into one line, because the command `./bin/pulsar sql-worker` can't identify the one line style now.

```
--add-opens
java.base/java.lang=ALL-UNNAMED
```

---

More information for you about the source code:

The command `./bin/pulsar sql-worker` will run the python launcher code `lib/presto/bin/launcher.py`. In `lib/presto/bin/launcher.py`, it uses the python method `os.execvpe` to execute the `java` command.

**Comment by tisonkun** [comment]
Fixed at 2.10.1.
